packages/core/src/app/customer/CheckoutButtonContainer.tsx

```diff
@@ -54,12 +55,17 @@ const CheckoutButtonContainer: FunctionComponent<CheckoutButtonContainerProps &
         return null;
     }
 
+    const { getConfig } = checkoutState.data;
+
     const renderButtons = () => availableMethodIds.map((methodId) => {
         if (isPaymentStepActive && isPayPalCommerce(methodId)) {
             return null;
         }
 
-        const ResolvedCheckoutButton = resolveCheckoutButton({ id: methodId });
+        const ResolvedCheckoutButton = resolveCheckoutButton(
+            { id: methodId },
+            isExperimentEnabled(getConfig()?.checkoutSettings, 'CHECKOUT-9432.lazy_load_payment_components', false)
```

Comment:
Should the fallback value be true here?

Reply:
The fallback value is set to `false` for the following reasons:
* Checkout microapp deployments are now decoupled from BCApp, but experiment flags are still released through BCApp. This means we can’t merge this PR until the relevant experiment flag has been released to all tiers.
* I plan to remove the experiment and the old code path in a week, so the impact on custom checkout after the experiment is removed will be minimal.

We should have a separate discussion on how to propagate experiment values going forward. Coupling them with BCApp causes the first problem mentioned above. Ideally, we’d be able to query the LD experiment directly on the client side. We need to investigate whether this is possible and what the performance impact would be.

Reply:
Yes, once BCApp PR reaches T3 and experiment is 100%, we can verify that the change is safe and we can proceed to remove the experiment and the old code paths.

Reply:
`isExperimentEnabled` is designed to handle situations where the experiment value is `undefined`. Here, we want different behaviours when the experiment is `undefined`, so we are not seeing the usual `true`.

> Ideally, we’d be able to query the LD experiment directly on the client side. 

As DC mentioned above, this solution is ideal, it will allow us to read newly set experiment values without touching BCapp code.